client/src/translations/index.tsx

```diff
@@ -0,0 +1,790 @@
+  Link,
+  useParams,
+  useSearchParams,
+} from "react-router-dom";
+import useSWR from "swr";
+import dayjs from "dayjs";
+import relativeTime from "dayjs/plugin/relativeTime";
+
+import "./index.scss";
+
+import { PageContentContainer } from "../ui/atoms/page-content";
+
+dayjs.extend(relativeTime);
+
+interface DocumentPopularity {
+  value: number;
+  ranking: number;
+  parentValue: number;
+  parentRanking: number;
+}
+
+type CountByType = { [key: string]: number };
+
+interface DocumentDifferences {
+  countByType: CountByType;
+  total: number;
+}
+
+interface DocumentEdits {
+  modified: string;
+  parentModified: string;
+  commitURL: string;
+  parentCommitURL: string;
+}
+
+interface Document {
+  mdn_url: string;
+  edits: DocumentEdits;
+  title: string;
+  popularity: DocumentPopularity;
+  differences: DocumentDifferences;
+}
+
+interface Counts {
+  found: number;
+  total: number;
+  pages: number;
+  noParents: number;
+  cacheMisses: number;
+}
+
+interface Times {
+  took: number;
+}
+
+interface FlawLevel {
+  name: string;
+  level: string;
+  ignored: boolean;
+}
+
+interface Locale {
+  locale: string;
+  language: {
+    English: string;
+    native: string;
+  };
+  isActive: boolean;
+}
+
+interface Data {
+  counts: Counts;
+  documents: Document[];
+  times: Times;
+  flawLevels: FlawLevel[];
+}
+
+interface LocalesData {
+  locales: Locale[];
+}
+
+interface LocaleStorageData {
+  lastLoadTime?: number;
+}
+interface StorageData {
+  [locale: string]: LocaleStorageData;
+}
+
+const LOCALSTORAGE_KEY = "translations-dashboard";
+
+function saveStorage(locale: string, data: LocaleStorageData) {
+  try {
+    const stored = JSON.parse(
+      localStorage.getItem(LOCALSTORAGE_KEY) || "{}"
+    ) as StorageData;
+    stored[locale] = Object.assign({}, stored[locale] || {}, data);
+    localStorage.setItem(LOCALSTORAGE_KEY, JSON.stringify(stored));
+  } catch (err) {
+    console.warn("Unable to save to localStorage", err);
+  }
+}
+
+function getStorage(locale: string): LocaleStorageData | null {
+  try {
+    const stored = JSON.parse(localStorage.getItem(LOCALSTORAGE_KEY) || "{}");
+    if (stored) {
+      return stored[locale] as LocaleStorageData;
+    }
+  } catch (err) {
+    console.warn("Unable to retrieve from localStorage", err);
+  }
+  return null;
+}
+
+export default function AllTranslations() {
+  const { locale } = useParams();
+
+  const [lastData, setLastData] = React.useState<Data | null>(null);
+
+  React.useEffect(() => {
+    let title = "All translations";
+    if (locale.toLowerCase() !== "en-us") {
+      title += ` for ${locale}`;
+    }
+    if (lastData) {
+      title = `(${lastData.counts.found.toLocaleString()} found) ${title}`;
+    }
+    document.title = title;
+  }, [lastData, locale]);
+
+  const getAPIUrl = React.useCallback(() => {
+    const params = createSearchParams({
+      locale,
+    });
+    return `/_translations?${params.toString()}`;
+  }, [locale]);
```

Comment:
I'd just turn that into a memo and use the value directly in all the `useSWRs`
```suggestion
  const apiURL = React.useMemo(() => {
    if (locale.toLowerCase() === "en-us") {
      return null;
    }
    const params = createSearchParams({
      locale,
    });
    return `/_translations?${params.toString()}`;
  }, [locale]);
```